pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -138,6 +139,19 @@ func validateStepOverrides(overrides []TaskRunStepOverride) (errs *apis.FieldErr
 	return errs
 }
 
+// validateStepOverridesResourcesRequirements() validates if both step-level and task-level resources requirements are configured
+func validateStepOverridesResourcesRequirements(taskSpec *TaskSpec, overrides []TaskRunStepOverride) (errs *apis.FieldError) {
```

Comment:
I think you also need to add TaskRun.resources to the api

Reply:
you mean `TaskRun.spec.resources` right? 
https://github.com/austinzhao-go/pipeline/blob/tep-0104-per-task-resources-requirements/pkg/apis/pipeline/v1beta1/taskrun_types.go#L44

rechecked our "API changes" section, think I understood wrongly as the referred `TaskSpec` field.

Reply:
yes exactly

Reply:
closed - added the field and related validations